Mlem/App/Models/MlemStats/MlemStats.swift

```diff
@@ -54,12 +54,26 @@ class MlemStats {
     }
     
     @MainActor
-    func searchInstances(query: String) async throws -> [InstanceSummary] {
+    func searchInstances(query: String, sort: InstanceSort = .score) async throws -> [InstanceSummary] {
         try await MlemStats.main.loadInstances()
```

Comment:
This logic is super weird--I must have missed it last review. This depends on the assumption that `searchInstances` will be called on `MlemStats.main`, since otherwise this `loadInstances` wouldn't do anything, but given that assumption it's redundant to call `MlemStats.main.loadInstances` here instead of a simple `loadInstances`. It's also a little odd to implicitly set `instances` with this call then filter and reset `instances` in the body of the function.

A cleaner way to handle this all would be:

- Extract the actual loading (L39-42 and associated error handling) into a `fetchInstances()` helper that returns `[InstanceSummary`
- Have `loadInstances` and `searchInstances` both call `fetchInstances`, then have `searchInstances` perform filtering on the response before assigning to `instances`
- Add guards to both methods that `self === MlemStats.main` with `assertionFailures` if not main

Alternatively, and I think possibly cleaner still:
- Move all loading work into `searchInstances` and make both `query` and `sort` optional
- Either delete `loadInstances` entirely or make it a simple passthrough to `searchInstances`that provides `nil` params
- Again, add `self === MlemStats.main` guard

Reply:
Yeah, makes sense. I think at some point I had this function outside of `MlemStats` and then moved it inside - I must have missed changing `MlemStats.main.loadInstances()` to just `loadInstances`. I'll fix it 👍 

> It's also a little odd to implicitly set instances with this call then filter and reset instances in the body of the function.

`instances` isn't changed in the body of `searchInstances` - `instances` is local to the function scope in this case.